include/cxx.h

```diff
@@ -156,6 +164,8 @@ class Slice final
   T *data() const noexcept;
   std::size_t size() const noexcept;
   std::size_t length() const noexcept;
+  T &operator[](std::size_t n) noexcept;
+  T &at(std::size_t n);
```

Comment:
These changes are not related to opaque type support so I've pulled them out to #598.